### ISSUE TITLE
rls-v2.5 benchdnn reorder on AArch64 failed

### ISSUE THREAD

**Issue by kawakami-k** [oneapi-src/oneDNN#1272]
# Summary

Two problems of `./benchdnn --reorder --batch=inputs/reorder/test_reorder_all` failed
(The master branch with my patch https://github.com/kawakami-k/oneDNN/tree/issue1256 for AArch64 also failed with these two problems.)

# Version

rls-v2.5
543b8f86f4659d36dddcff1d2bddb810d3740889

# Environment

## Mac Mini (M1)
* CPU: M1
* OS: macOS Monterey 12.2
* cmake: 3.19.2
* Compiler: Apple clang version 12.0.5 (clang=1205.0.22.9)

## FX700 (A64FX), 
* CPU: A64FX
* OS: CentOS 8
* cmake: 3.11.4
* gcc: 8.3.1 20190507 (Red Hat 8.3.1-4)

# Steps to reproduce

```
./benchdnn --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx4b --attr-oscale=common:4.29497e+09 1x17x9x5
./benchdnn --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx8b --attr-oscale=common:4.29497e+09 1x17x9x5
```

# Observed behavior

```
[ 721][arg:17][0,17,0,0,0,0] fp:  0.f dt:     -nan 
[ 725][arg:17][0,17,0,1,0,0] fp:  0.f dt:     -nan 
[ 729][arg:17][0,17,0,2,0,0] fp:  0.f dt:     -nan 
[ 733][arg:17][0,17,0,3,0,0] fp:  0.f dt:     -nan 
[ 737][arg:17][0,17,0,4,0,0] fp:  0.f dt:     -nan 
[ 741][arg:17][0,17,1,0,0,0] fp:  0.f dt:     -nan 
[ 745][arg:17][0,17,1,1,0,0] fp:  0.f dt:     -nan 
[ 749][arg:17][0,17,1,2,0,0] fp:  0.f dt:     -nan 
[ 753][arg:17][0,17,1,3,0,0] fp:  0.f dt:     -nan 
@@@ [arg:17] check_zero_padding failed
21982:UNTESTED_FAILED (errors:0 total:0) __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx4b --attr-oscale=common:4.29497e+09 1x17x9x5
[ 721][arg:17][0,17,0,0,0,0] fp:  0.f dt:     -nan 
[ 729][arg:17][0,17,0,1,0,0] fp:  0.f dt:     -nan 
[ 737][arg:17][0,17,0,2,0,0] fp:  0.f dt:     -nan 
[ 745][arg:17][0,17,0,3,0,0] fp:  0.f dt:     -nan 
[ 753][arg:17][0,17,0,4,0,0] fp:  0.f dt:     -nan 
[ 761][arg:17][0,17,1,0,0,0] fp:  0.f dt:     -nan 
[ 769][arg:17][0,17,1,1,0,0] fp:  0.f dt:     -nan 
[ 777][arg:17][0,17,1,2,0,0] fp:  0.f dt:     -nan 
[ 785][arg:17][0,17,1,3,0,0] fp:  0.f dt:     -nan 
@@@ [arg:17] check_zero_padding failed
21983:UNTESTED_FAILED (errors:0 total:0) __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx8b --attr-oscale=common:4.29497e+09 1x17x9x5
```

**Comment by mgouicem** [comment]
Thanks for the report @kawakami-k . Could you send the output of `ONEDNN_VERBOSE=1` for this test, with both the failing version, and the last known working version?

**Comment by kawakami-k** [comment]
Hi, @mgouicem 

It seems the failure occured after the following commit, which was pushed after v2.2-branch cut off. So the two test patterns also failed at v2.3 and v2.4.

```
commit bb0f475615097915a17cb0a5cd4a7980d1e84691
Author: Robert Dobrowolski <robert.dobrowolski@intel.com>
Date:   Wed Mar 10 07:57:45 2021 +0100

    cpu: reorder: remove zero padding macro from reference reorder
```

### v2.5 the first failing version 5917860513ec94274a157d9642695393858cd205
```
onednn_verbose,info,oneDNN v2.5.2 (commit 5917860513ec94274a157d9642695393858cd205)
onednn_verbose,info,cpu,runtime:OpenMP
onednn_verbose,info,cpu,isa:AArch64 SVE (512 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,62.646
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd4b:f0,attr-oscale:0:4294969856.000000 ,,1x17x9x5,30.094
[ 721][arg:17][0,17,0,0,0,0] fp:  0.f dt:     -nan 
[ 725][arg:17][0,17,0,1,0,0] fp:  0.f dt:     -nan 
[ 729][arg:17][0,17,0,2,0,0] fp:  0.f dt:     -nan 
[ 733][arg:17][0,17,0,3,0,0] fp:  0.f dt:     -nan 
[ 737][arg:17][0,17,0,4,0,0] fp:  0.f dt:     -nan 
[ 741][arg:17][0,17,1,0,0,0] fp:  0.f dt:     -nan 
[ 745][arg:17][0,17,1,1,0,0] fp:  0.f dt:     -nan 
[ 749][arg:17][0,17,1,2,0,0] fp:  0.f dt:     -nan 
[ 753][arg:17][0,17,1,3,0,0] fp:  0.f dt:     -nan 
@@@ [arg:17] check_zero_padding failed
@@@ error [int execute_and_wait(perf_function_t&, dnnl_engine* const&, const args_t&):161]: 'check_zero_padding(args.dnn_mem(i), args.arg(i))' -> 1
@@@ error [int reorder::doit(const reorder::prb_t*, res_t*):422]: 'execute_and_wait(prim, args)' -> 1
0:UNTESTED_FAILED (errors:0 total:0) __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx4b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 failed:1 listed:0
total compute_ref: sum(s):0.00
onednn_verbose,info,oneDNN v2.5.2 (commit 5917860513ec94274a157d9642695393858cd205)
onednn_verbose,info,cpu,runtime:OpenMP
onednn_verbose,info,cpu,isa:AArch64 SVE (512 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,63.5762
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd8b:f0,attr-oscale:0:4294969856.000000 ,,1x17x9x5,39.8911
[ 721][arg:17][0,17,0,0,0,0] fp:  0.f dt:     -nan 
[ 729][arg:17][0,17,0,1,0,0] fp:  0.f dt:     -nan 
[ 737][arg:17][0,17,0,2,0,0] fp:  0.f dt:     -nan 
[ 745][arg:17][0,17,0,3,0,0] fp:  0.f dt:     -nan 
[ 753][arg:17][0,17,0,4,0,0] fp:  0.f dt:     -nan 
[ 761][arg:17][0,17,1,0,0,0] fp:  0.f dt:     -nan 
[ 769][arg:17][0,17,1,1,0,0] fp:  0.f dt:     -nan 
[ 777][arg:17][0,17,1,2,0,0] fp:  0.f dt:     -nan 
[ 785][arg:17][0,17,1,3,0,0] fp:  0.f dt:     -nan 
@@@ [arg:17] check_zero_padding failed
@@@ error [int execute_and_wait(perf_function_t&, dnnl_engine* const&, const args_t&):161]: 'check_zero_padding(args.dnn_mem(i), args.arg(i))' -> 1
@@@ error [int reorder::doit(const reorder::prb_t*, res_t*):422]: 'execute_and_wait(prim, args)' -> 1
0:UNTESTED_FAILED (errors:0 total:0) __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx8b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 failed:1 listed:0
total compute_ref: sum(s):0.00
```

### v2.5 the last working version ff7ae00c074ea09048bcc27f905385d1f6a6b830
```
onednn_verbose,info,oneDNN v2.5.2 (commit ff7ae00c074ea09048bcc27f905385d1f6a6b830)
onednn_verbose,info,cpu,runtime:OpenMP
onednn_verbose,info,cpu,isa:AArch64 SVE (512 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,80.2551
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd4b:f0,attr-oscale:0:4294969856.000000 ,,1x17x9x5,69.5549
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd4b:f0 dst_f32::blocked:abcd:f0,,,1x17x9x5,39.7529
0:PASSED __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx4b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:1 skipped:0 mistrusted:0 unimplemented:0 failed:0 listed:0
total compute_ref: sum(s):0.00
onednn_verbose,info,oneDNN v2.5.2 (commit ff7ae00c074ea09048bcc27f905385d1f6a6b830)
onednn_verbose,info,cpu,runtime:OpenMP
onednn_verbose,info,cpu,isa:AArch64 SVE (512 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,94.084
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd8b:f0,attr-oscale:0:4294969856.000000 ,,1x17x9x5,64.291
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd8b:f0 dst_f32::blocked:abcd:f0,,,1x17x9x5,35.949
0:PASSED __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx8b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:1 skipped:0 mistrusted:0 unimplemented:0 failed:0 listed:0
total compute_ref: sum(s):0.00
```

### v2.4 the first failing version bb0f475615097915a17cb0a5cd4a7980d1e84691
```
dnnl_verbose,info,oneDNN v2.2.0 (commit bb0f475615097915a17cb0a5cd4a7980d1e84691)
dnnl_verbose,info,cpu,runtime:OpenMP
dnnl_verbose,info,cpu,isa:AArch64 SVE (512 bits)
dnnl_verbose,info,gpu,runtime:none
dnnl_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,3.05396
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd4b:f0,oscale:0:4.29497e+09;,,1x17x9x5,0.802002
[ 721][arg:17][0,17,0,0,0,0] fp:  0.f dt:     -nan 
[ 725][arg:17][0,17,0,1,0,0] fp:  0.f dt:     -nan 
[ 729][arg:17][0,17,0,2,0,0] fp:  0.f dt:     -nan 
[ 733][arg:17][0,17,0,3,0,0] fp:  0.f dt:     -nan 
[ 737][arg:17][0,17,0,4,0,0] fp:  0.f dt:     -nan 
[ 741][arg:17][0,17,1,0,0,0] fp:  0.f dt:     -nan 
[ 745][arg:17][0,17,1,1,0,0] fp:  0.f dt:     -nan 
[ 749][arg:17][0,17,1,2,0,0] fp:  0.f dt:     -nan 
[ 753][arg:17][0,17,1,3,0,0] fp:  0.f dt:     -nan 
@@@ [arg:17] check_zero_padding failed
0:UNTESTED_FAILED (errors:0 total:0) __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx4b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 failed:1 listed:0
dnnl_verbose,info,oneDNN v2.2.0 (commit bb0f475615097915a17cb0a5cd4a7980d1e84691)
dnnl_verbose,info,cpu,runtime:OpenMP
dnnl_verbose,info,cpu,isa:AArch64 SVE (512 bits)
dnnl_verbose,info,gpu,runtime:none
dnnl_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,2.73608
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd8b:f0,oscale:0:4.29497e+09;,,1x17x9x5,0.908936
[ 721][arg:17][0,17,0,0,0,0] fp:  0.f dt:     -nan 
[ 729][arg:17][0,17,0,1,0,0] fp:  0.f dt:     -nan 
[ 737][arg:17][0,17,0,2,0,0] fp:  0.f dt:     -nan 
[ 745][arg:17][0,17,0,3,0,0] fp:  0.f dt:     -nan 
[ 753][arg:17][0,17,0,4,0,0] fp:  0.f dt:     -nan 
[ 761][arg:17][0,17,1,0,0,0] fp:  0.f dt:     -nan 
[ 769][arg:17][0,17,1,1,0,0] fp:  0.f dt:     -nan 
[ 777][arg:17][0,17,1,2,0,0] fp:  0.f dt:     -nan 
[ 785][arg:17][0,17,1,3,0,0] fp:  0.f dt:     -nan 
@@@ [arg:17] check_zero_padding failed
0:UNTESTED_FAILED (errors:0 total:0) __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx8b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 failed:1 listed:0
```

### v2.4 the last working version f4f8d3e96783e39c1ef6347a79d07dea0e510471
```
dnnl_verbose,info,oneDNN v2.2.0 (commit f4f8d3e96783e39c1ef6347a79d07dea0e510471)
dnnl_verbose,info,cpu,runtime:OpenMP
dnnl_verbose,info,cpu,isa:AArch64 SVE (512 bits)
dnnl_verbose,info,gpu,runtime:none
dnnl_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,3.6311
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd4b:f0,oscale:0:4.29497e+09;,,1x17x9x5,1.6521
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd4b:f0 dst_f32::blocked:abcd:f0,,,1x17x9x5,0.410156
0:PASSED __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx4b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:1 skipped:0 mistrusted:0 unimplemented:0 failed:0 listed:0
dnnl_verbose,info,oneDNN v2.2.0 (commit f4f8d3e96783e39c1ef6347a79d07dea0e510471)
dnnl_verbose,info,cpu,runtime:OpenMP
dnnl_verbose,info,cpu,isa:AArch64 SVE (512 bits)
dnnl_verbose,info,gpu,runtime:none
dnnl_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,4.84985
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd8b:f0,oscale:0:4.29497e+09;,,1x17x9x5,1.95703
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd8b:f0 dst_f32::blocked:abcd:f0,,,1x17x9x5,0.873047
0:PASSED __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx8b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:1 skipped:0 mistrusted:0 unimplemented:0 failed:0 listed:0
```


### v2.3 the first failing version bb0f475615097915a17cb0a5cd4a7980d1e84691
```
dnnl_verbose,info,oneDNN v2.2.0 (commit bb0f475615097915a17cb0a5cd4a7980d1e84691)
dnnl_verbose,info,cpu,runtime:OpenMP
dnnl_verbose,info,cpu,isa:AArch64 SVE (512 bits)
dnnl_verbose,info,gpu,runtime:none
dnnl_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,2.79297
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd4b:f0,oscale:0:4.29497e+09;,,1x17x9x5,0.450928
[ 721][arg:17][0,17,0,0,0,0] fp:  0.f dt:     -nan 
[ 725][arg:17][0,17,0,1,0,0] fp:  0.f dt:     -nan 
[ 729][arg:17][0,17,0,2,0,0] fp:  0.f dt:     -nan 
[ 733][arg:17][0,17,0,3,0,0] fp:  0.f dt:     -nan 
[ 737][arg:17][0,17,0,4,0,0] fp:  0.f dt:     -nan 
[ 741][arg:17][0,17,1,0,0,0] fp:  0.f dt:     -nan 
[ 745][arg:17][0,17,1,1,0,0] fp:  0.f dt:     -nan 
[ 749][arg:17][0,17,1,2,0,0] fp:  0.f dt:     -nan 
[ 753][arg:17][0,17,1,3,0,0] fp:  0.f dt:     -nan 
@@@ [arg:17] check_zero_padding failed
0:UNTESTED_FAILED (errors:0 total:0) __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx4b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 failed:1 listed:0
dnnl_verbose,info,oneDNN v2.2.0 (commit bb0f475615097915a17cb0a5cd4a7980d1e84691)
dnnl_verbose,info,cpu,runtime:OpenMP
dnnl_verbose,info,cpu,isa:AArch64 SVE (512 bits)
dnnl_verbose,info,gpu,runtime:none
dnnl_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,2.77783
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd8b:f0,oscale:0:4.29497e+09;,,1x17x9x5,0.803955
[ 721][arg:17][0,17,0,0,0,0] fp:  0.f dt:     -nan 
[ 729][arg:17][0,17,0,1,0,0] fp:  0.f dt:     -nan 
[ 737][arg:17][0,17,0,2,0,0] fp:  0.f dt:     -nan 
[ 745][arg:17][0,17,0,3,0,0] fp:  0.f dt:     -nan 
[ 753][arg:17][0,17,0,4,0,0] fp:  0.f dt:     -nan 
[ 761][arg:17][0,17,1,0,0,0] fp:  0.f dt:     -nan 
[ 769][arg:17][0,17,1,1,0,0] fp:  0.f dt:     -nan 
[ 777][arg:17][0,17,1,2,0,0] fp:  0.f dt:     -nan 
[ 785][arg:17][0,17,1,3,0,0] fp:  0.f dt:     -nan 
@@@ [arg:17] check_zero_padding failed
0:UNTESTED_FAILED (errors:0 total:0) __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx8b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 failed:1 listed:0
```

### v2.3 the last working version 1681049bad80159ee5200b7ba8880b815488cea1
```
dnnl_verbose,info,oneDNN v2.2.0 (commit 1681049bad80159ee5200b7ba8880b815488cea1)
dnnl_verbose,info,cpu,runtime:OpenMP
dnnl_verbose,info,cpu,isa:AArch64 SVE (512 bits)
dnnl_verbose,info,gpu,runtime:none
dnnl_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,3.66284
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd4b:f0,oscale:0:4.29497e+09;,,1x17x9x5,1.81787
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd4b:f0 dst_f32::blocked:abcd:f0,,,1x17x9x5,0.409912
0:PASSED __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx4b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:1 skipped:0 mistrusted:0 unimplemented:0 failed:0 listed:0
dnnl_verbose,info,oneDNN v2.2.0 (commit 1681049bad80159ee5200b7ba8880b815488cea1)
dnnl_verbose,info,cpu,runtime:OpenMP
dnnl_verbose,info,cpu,isa:AArch64 SVE (512 bits)
dnnl_verbose,info,gpu,runtime:none
dnnl_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd16b:f0,,,1x17x9x5,4.00098
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd16b:f0 dst_f32:p:blocked:aBcd8b:f0,oscale:0:4.29497e+09;,,1x17x9x5,1.77197
dnnl_verbose,exec,cpu,reorder,simple:any,undef,src_f32:p:blocked:aBcd8b:f0 dst_f32::blocked:abcd:f0,,,1x17x9x5,0.900146
0:PASSED __REPRO: --reorder --sdt=f32 --ddt=f32 --stag=aBx16b --dtag=aBx8b --attr-oscale=common:4.29497e+09 1x17x9x5
tests:1 passed:1 skipped:0 mistrusted:0 unimplemented:0 failed:0 listed:0
```

**Comment by mgouicem** [comment]
Thanks for bisecting the issue. Looking into it.

**Comment by johnnynunez** [comment]
Can this make Neural Engine or GPU compatible with pytorch?

**Comment by kawakami-k** [comment]
Hi, @johnnynunez 
When oneDNN is built on Apple M1, oneDNN for AArch64 CPU is built. This oneDNN runs on the M1 CPU core. I have not tried it with Neural Engine.

**Comment by johnnynunez** [comment]
> Hi, @johnnynunez When oneDNN is built on Apple M1, oneDNN for AArch64 CPU is built. This oneDNN runs on the M1 CPU core. I have not tried it with Neural Engine.

And does it work with the gpu, or only with the cpu?

**Comment by igorsafo** [comment]
Reopening since it is not backported to rls-v2.5 and rls-v2.6 yet:
- #1346
- #1347

**Comment by igorsafo** [comment]
@kawakami-k The fixes were backported to rls-v2.5 and rls-v2.6 branches

**Comment by kawakami-k** [comment]
> @kawakami-k The fixes were backported to rls-v2.5 and rls-v2.6 branches

@igorsafo 
That's great. Thank you so much.

**Comment by igorsafo** [comment]
Clossing since the fix was promoted:
- https://github.com/oneapi-src/oneDNN/pull/1335
- https://github.com/oneapi-src/oneDNN/pull/1346
- https://github.com/oneapi-src/oneDNN/pull/1347
